ulm-semantics/test/execution.md

```diff
@@ -199,6 +208,100 @@ module ULM-TEST-EXECUTION
+                        | evaluatedStorageKey(Bytes)
+    rule storageKey(StorageName:String, Args:EncodeArgs)
+        => storageKey
+            ( concat
+                (   let buffer_id = :: bytes_hooks :: empty( .CallParamsList );
+                    .NonEmptyStatements
+                ,   codegenValuesEncoder
+                        ( buffer_id
+                        , (StorageName : str , encodeArgsToEncodeValues(Args))
+                        )
+                )
+            )
+    rule storageKey(Statements:NonEmptyStatements)
+        =>  storageKey({.InnerAttributes Statements buffer_id })
+    rule storageKey(ptrValue(_, u64(BytesId))) => storageKey(ulmBytesId(BytesId))
+    rule storageKey(ulmBytesValue(B:Bytes)) => evaluatedStorageKey(B)
+
+    syntax EventSignature ::= evaluatedEventSignature(Int)
+    rule eventSignature(Signature:String)
+        => evaluatedEventSignature(Bytes2Int(Keccak256raw(String2Bytes(Signature)), BE, Unsigned))
+
+    syntax EncodeValues ::= encodeValues(NonEmptyStatementsOrError)
+                          | encodeValues(Expression)  [strict(1)]
+                          | encodeValues(UlmExpression)  [strict(1)]
+                          | encodedValues(Bytes)
+    rule encodeValues(Args:EncodeArgs)
+        => encodeValues
+            ( concat
+                (   let buffer_id = :: bytes_hooks :: empty( .CallParamsList );
+                    .NonEmptyStatements
+                ,   codegenValuesEncoder
+                        ( buffer_id
+                        , encodeArgsToEncodeValues(Args)
+                        )
+                )
+            )
+    rule encodeValues(Statements:NonEmptyStatements)
+        =>  encodeValues({.InnerAttributes Statements buffer_id })
+    rule encodeValues(ptrValue(_, u64(BytesId))) => encodeValues(ulmBytesId(BytesId))
+    rule encodeValues(ulmBytesValue(B:Bytes)) => encodedValues(B)
+
+    rule SetAccountStorageHook(evaluatedStorageKey(B:Bytes), ptrValue(_, Value:Value))
+        => #SetAccountStorageHook
+                ( Bytes2Int(Keccak256raw(B), BE, Unsigned)
+                , valueToInteger(Value)
+                )
+    rule mock #SetAccountStorageHook(Key:Int, Value:Int) Result:UlmHookResult
+        => mock
+            SetAccountStorageHook(Key, Value)
+            Result
+    rule list_mock #SetAccountStorageHook(Key:Int, Value:Int) Result:UlmHookResult
+        => list_mock
+            SetAccountStorageHook(Key, Value)
+            Result
+
+    rule GetAccountStorageHook(evaluatedStorageKey(B:Bytes))
+        => #GetAccountStorageHook(Bytes2Int(Keccak256raw(B), BE, Unsigned))
+    rule mock #GetAccountStorageHook(Key:Int) Result:UlmHookResult
+        => mock
+            GetAccountStorageHook(Key)
+            Result
+    rule list_mock #GetAccountStorageHook(Key:Int) Result:UlmHookResult
+        => list_mock
+            GetAccountStorageHook(Key)
+            Result
```

Comment:
The use of overloaded names `Set/GetAccountStorageHook` and `#Set/GetAccountStorageHook` both in this file and in `ulm-semantics/main/hooks/ulm.md` makes the logic a bit hard to follow in these rules.

It looks in this file the declared sharp versions are intermediate forms --- perhaps they could be renamed to avoid conflict with the overloaded version in `ulm-semantics/main/hooks/ulm.md`?

I understand that the non-sharp version in this file is designed to make the test files look like the underlying functions in the semantics, so the overloading there seems to have a potentially bigger payoff in test readability.

Reply:
Renamed.